istioctl/cmd/workload.go

```diff
@@ -0,0 +1,402 @@
+This includes a MeshConfig resource, the cluster.env file, and necessary certificates and security tokens.
+Configure requires either the WorkloadGroup artifact path or its location on the API server.`,
+		Example: `# configure example using a local WorkloadGroup artifact
+configure -f workloadgroup.yaml -o config
+
+# configure example using the API server
+configure --name foo --namespace bar -o config`,
+		Args: func(cmd *cobra.Command, args []string) error {
+			if filename == "" && (name == "" || namespace == "") {
+				return fmt.Errorf("expecting a WorkloadGroup artifact file or the name and namespace of an existing WorkloadGroup")
+			}
+			if outputDir == "" {
+				return fmt.Errorf("expecting an output directory")
+			}
+			if clusterID == "" {
+				return fmt.Errorf("expecting a cluster id")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			wg := &clientv1alpha3.WorkloadGroup{}
+			kubeClient, err := kube.NewExtendedClient(kube.BuildClientCmd(kubeconfig, configContext), "")
+			if err != nil {
+				return err
+			}
+			if filename != "" {
+				if err := readWorkloadGroup(filename, wg); err != nil {
+					return err
+				}
+			} else {
+				wg, err = kubeClient.Istio().NetworkingV1alpha3().WorkloadGroups(namespace).Get(context.Background(), name, metav1.GetOptions{})
+				// errors if the requested workload group does not exist in the given namespace
+				if err != nil {
+					return fmt.Errorf("workloadgroup %s not found in namespace %s: %v", name, namespace, err)
+				}
+			}
+
+			if err = createConfig(kubeClient, wg, clusterID, revision, outputDir); err != nil {
+				return err
+			}
+			return nil
+		},
+	}
+	configureCmd.PersistentFlags().StringVarP(&filename, "file", "f", "", "filename of the WorkloadGroup artifact")
+	configureCmd.PersistentFlags().StringVar(&name, "name", "", "The name of the workload group")
+	configureCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", "", "The namespace that the workload instances belongs to")
+	configureCmd.PersistentFlags().StringVarP(&outputDir, "output", "o", "", "Output directory for generated files")
+	configureCmd.PersistentFlags().StringVar(&clusterID, "clusterID", "", "The ID used to identify the cluster")
+	configureCmd.PersistentFlags().Int64Var(&tokenDuration, "tokenDuration", 3600, "The token duration in seconds (default: 1 hour)")
+	configureCmd.PersistentFlags().StringVar(&revision, "revision", "", "control plane revision (experimental)")
+	return configureCmd
+}
+
+func readWorkloadGroup(filename string, wg *clientv1alpha3.WorkloadGroup) error {
+	f, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	if err = yaml.Unmarshal(f, wg); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Creates all the relevant config for the given workload group and cluster
+func createConfig(kubeClient kube.ExtendedClient, wg *clientv1alpha3.WorkloadGroup, clusterID, revision, outputDir string) error {
```

Comment:
Can definitely be a followup: Right now once the command is run, nothing happens. I would recommend, at the very least, a line printed like `wrote out config to <DIR>`, and Ideally a bit about next steps (see istio.io/vm-setup for next steps, or whatever)